external/fv3fit/tests/sklearn/test_transformer.py

```diff
@@ -0,0 +1,96 @@
+from fv3fit.sklearn.transformer import SkTransformer
+import numpy as np
+import os
+from sklearn.preprocessing import StandardScaler
+
+
+class DummyTransformer:
+    def __init__(self, latent_dim, output_dim, encoded_constant, decoded_constant):
+        self.latent_dim = latent_dim
+        self.output_dim = output_dim
+        self.encoded_constant = encoded_constant
+        self.decoded_constant = decoded_constant
+
+    def transform(self, X):
+        return np.ones(self.latent_dim) + self.encoded_constant
+
+    def inverse_transform(self, X):
+        return np.ones(self.output_dim) + self.decoded_constant
+
+
+def test_sktransformer_roundtrip():
+    latent_dim = 5
+    output_dim = 10
+    n_vars = 3
+    encoded_constant = 0
+    decoded_constant = -2
+
+    scaler = StandardScaler()
+    scaler.fit(np.random.rand(5, n_vars * output_dim))
+    transformer = transformer = DummyTransformer(
+        latent_dim, n_vars * output_dim, encoded_constant, decoded_constant
+    )
+    sktransformer = SkTransformer(transformer, scaler, enforce_positive_outputs=False)
+
+    x = [np.random.rand(output_dim) for i in range(n_vars)]
+    target = np.ones(n_vars * output_dim) + decoded_constant
+    outputs = sktransformer.predict(x)
+
+    denormed_target = sktransformer.scaler.inverse_transform(target.reshape(1, -1))
+    np.testing.assert_allclose(np.concatenate(outputs, axis=-1), denormed_target)
+
+
+def test_sktransformer_enforce_positive():
+    latent_dim = 5
+    output_dim = 3
+    n_vars = 2
+    encoded_constant = 0
+    decoded_constant = -5
+
+    scaler = StandardScaler()
+    scaler.fit(np.random.rand(5, n_vars * output_dim))
+    transformer = transformer = DummyTransformer(
+        latent_dim, n_vars * output_dim, encoded_constant, decoded_constant
+    )
+    sktransformer = SkTransformer(transformer, scaler, enforce_positive_outputs=False)
+    sktransformer_enforce_positive = SkTransformer(
+        transformer, scaler, enforce_positive_outputs=True
+    )
+
+    x = [np.arange(output_dim) * 10.0, np.arange(output_dim) * -10.0]
+    outputs = np.concatenate(sktransformer.predict(x), axis=-1).reshape(-1)
```

Comment:
Instead of all the shape manipulation, can we just check that the split outputs are equal for for the first set of features and `0.0` for the second?

Reply:
Simplified the test so one variable is all negative and other is all positive